bootstrap/pkg/kfapp/coordinator/coordinator.go

```diff
@@ -697,6 +697,23 @@ func LoadKfAppCfgFile(cfgfile string) (kftypesv3.KfApp, error) {
 		}
 	}
 
+	// Since we know we have a local file we can set a default name if none is set based on the local directory
+	if kfdef.Name == "" {
+		absAppPath, err := filepath.Abs(appFile)
+
+		if err != nil {
+			return nil, &kfapis.KfError{
+				Code:    int(kfapis.INTERNAL_ERROR),
+				Message: fmt.Sprintf("KfDef.Name isn't set and there was a problem inferring the name based on the path %v; error: %v\nPlease set the name explicitly in the KFDef spec.", appFile, err),
+			}
+		}
+
+		appDir := filepath.Dir(absAppPath)
+
+		kfdef.Name = filepath.Base(appDir)
```

Comment:
If the app file is located directly under root (e.g. "~/app.yaml") this means the kfdef name is now "~". Will this cause problems?

Reply:
Good catch. I correctly handle that now. If the name ends up being "" we throw and error.
PTAL